src/backend/utils.ts

```diff
@@ -342,6 +344,25 @@ ${isLinux ? `PROTOCOL: ${xEnv}` : ''}`
   return systemInfoCache
 }
 
+const getSystemInfo = async (): Promise<string> => {
+  try {
```

Comment:
at the beginning of `getsustemInfoInternal` there's an if to use a cached value for the info to prevent extra work, I wonder if we should move that logic here instead, we don't really need to do the try/catch promises etc if we have that information cached

I mean this block:

```
  if (systemInfoCache !== '') {
    return systemInfoCache
  }
```

Reply:
Good idea! You save the whole promise